services/automerge/pull_auto_merge.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2019 Gitea. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package automerge
+
+import (
+	"strings"
+
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/git"
+	"code.gitea.io/gitea/modules/log"
+	pullservice "code.gitea.io/gitea/services/pull"
+)
+
+// This package merges a previously scheduled pull request on successful status check.
+// It is a separate package to avoid cyclic dependencies.
+
+// MergeScheduledPullRequest merges a previously scheduled pull request when all checks succeeded
+// Maybe FIXME: Move the whole check this function does into a separate go routine and just ping from here?
+func MergeScheduledPullRequest(sha string, repo *models.Repository) (err error) {
+	// First, get the branch associated with that commit sha
+	r, err := git.OpenRepository(repo.RepoPath())
+	if err != nil {
+		return
+	}
+	defer r.Close()
+	commitID := git.MustIDFromString(sha)
+	tree := git.NewTree(r, commitID)
+	commit := &git.Commit{
+		Tree: *tree,
+		ID:   commitID,
+	}
+	branches, err := commit.GetBranchNames()
+	if err != nil {
+		return
+	}
+
+	for _, branch := range branches {
+		// We get the branch name with a \n at the end which is not in the db so we strip it out
+		branch = strings.Trim(branch, "\n")
+		// Then get all prs for that branch
+		pr, err := models.GetPullRequestByHeadBranch(branch, repo)
```

Comment:
The repo here is not the headRepo if the pr is from forked repo.

Reply:
That's because the commit is not on the head branch but on a copyed branch named as as ``pull/*/head``

Reply:
But the checks are set for the head repo

Reply:
Sadly the repo used here is not the head repo of pr, but the repo that the commit belongs to. So if you create a pr from a forked repo, it's wrong... 

ref:
https://github.com/kolaente/gitea/blob/f2a7024e62afa6d4d19815e489c4b47ff0492d94/modules/repofiles/commit_status.go#L18

https://github.com/kolaente/gitea/blob/f2a7024e62afa6d4d19815e489c4b47ff0492d94/modules/repofiles/commit_status.go#L42

Reply:
> Sadly the repo used here is not the head repo of pr, but the repo that the commit belongs to.

That makes sense. I'll take a look.

Reply:
Okay so it seems like the pr would just not get merged in that case: 
![image](https://user-images.githubusercontent.com/13721712/84694362-5dd5fd80-af49-11ea-9f61-70309387cf50.png)

I guess I'll need to find a way to reliably get the pr to that branch.

Reply:
It works now! (Was quite a ride and some discussion on discord went into this)